### DISCUSSION TITLE
Uploading via WebDAV changes modified date of files

### DISCUSSION THREAD

**Discussion by atvking** [photoprism/photoprism#2550] [Q&A]
I'm running PhotoPrism in a docker container on an UnRaid server and it doesn't seem to matter what client I use (FolderSync Pro, PhotoSync, Windows File Explorer, etc.) to upload to the Originals folder, the modification date of the file is always changed to the date/time of upload rather than the date/time the photo was taken. Is there some way to avoid this?

Uploading the same files using the same clients to a folder in Nextcloud via its WebDAV server doesn't have the same issue.

I'm trying to avoid uploading to a Nextcloud folder via Nextclouds WebDAV and setting my Originals folder to use that folder because then I would have to schedule a cron job (or something) to index the files, whereas they are indexed automatically when using Photoprisms WebDAV.

I have already tried setting the PHOTOPRISM_DISABLE_CHOWN flag but I didn't have much hope that it would work anyway because the folder ownership and permissions are identical between the Originals folder on my PhotoPrism share and the Nextcloud folder I am syncing the same files to (as a test) on my Nextcloud share and do not change after uploading a file.

Suggestions?

**Comment by lastzero** [comment]
That's expected. An uploaded file effectively is a new file for the server and I'm currently not aware of any mod time information the comes along with it when using WebDAV. At best rsync might preserve it. Why is the mod time important to you? Any standard (besides what Nextcloud does) that says the mod time of uploads must be preserved or match the metadata? I hear this the first time...

**Reply by atvking** [reply]
Interesting...

It's important because if I ever access the photos or videos from another client (Nextcloud, windows explorer, or anywhere really), download them, or share them with someone else then there's a fairly good chance that the files will be ordered by modification date in the event that you would like to order them from newest to oldest (or vice versa). If I am unable to sync my photos for an extended period of time then I will end up with photos that all appear to have been taken on the same day, even if in reality they were taken at different times across multiple weeks. Rsync does preserve the mod dates and in fact, it sees the files as different between my PhotoPrism folder and equivalent Nextcloud folder because of the differences in mod dates even though the file name and size are the same.

I'm not sure how Nextcloud implements this functionality, but it would certainly be nice to have the option to preserve the mod date.

For now, I suppose I will resort to using the Nextcloud WebDAV and triggering the indexing on the originals folder manually (or via script).

**Reply by lastzero** [reply]
Everyone who wants this is invited to contribute, starting with research on how it could be implemented and what is required to make it work (client and server side).

**Reply by atvking** [reply]
That makes sense, thanks!

**Reply by Daniel-PT** [reply]
Ouh i did also test this and yes the date of file is getting modyfied to today :(

**Reply by lastzero** [reply]
Help welcome (we can also create a second, similar issue for WebDAV if needed):
- https://github.com/photoprism/photoprism/issues/3942

**Reply by Daniel-PT** [reply]
@lastzero i think that will be super nice :)

**Reply by graciousgrey** [reply]
:point_right: https://github.com/photoprism/photoprism/issues/3959

**Comment by phpbg** [comment]
Hi, FYI nextcloud uses X-OC-MTime and X-OC-CTime headers (it is not standard in webdav protocol)

See https://docs.nextcloud.com/server/latest/developer_manual/client_apis/WebDAV/basic.html#request-headers
